### PR TITLE
Save configuration & command-based IP listing.

### PR DESCRIPTION
**Pull request by GuzioMG**
So. Guess what. I'M BACK! And my new repo is with me. After 2 fails (#578, which was rejected and #610, which I closed myself), I'm here again. So, on #610 I told, I'll come with a refreshed repository, re-compatibilized (if this word exists), but keeping old features. Help is already in master, so I'm not cloning it, because just why? Next, ipconfig. I matched it with a new permission system, and it's working awesome! And now the trickiest part. My weird exit system. I decided not to ask on exit, but on save. So here is actually a new feature, which you've never seen before on any of my PR: configurable save system. I'm not going to describe what it does. You're reviving code anyway. Just look at ServerCommands\SaveCommand.cs and ServerCommands\ExitCommand.cs to you'll see, what I mean.
> When on #610 I said, I'll be back soon, I didn't think THAT soon!

By the way, I used gitpod.io and some parts of the code may be hard to read - InteliSense in gitpod.io is terrible! I'm not going to use it anymore.......

### CODE REVIEW

**Comment by Ghaarg** [comment]
Hey there,
just a little comment from my experience. If you commit your changes from your master, then you will have new cyclic commits as soon as your PR is merged and you update your fork.
Best action imo would be to branch your current progress. 
Then hard reset your master to the upstream version and force push it.
Make sure to backup your data if something goes wrong.

**Comment by GuzioMG** [comment]
OK, thank you. I'm going to wait until this to merge and then start using your practice.

**Comment by GuzioMG** [comment]
Is this even alive? This PR is extremely old and only @Ghaarg suggested to branch progress. Damn, wake up somebody! There are 6 more PRs from April, 2 with review and only one can't be merged. And no one seems to care! Did this repo died or, what? The last commit is from 22 Apr. What's going on, guys?

**Comment by Coding-Hen** [comment]
@GuzioMG I know what you mean, its a shame nothing is moving at the moment within the discord apparently the devs are taking a break. @Sunrunner37 Are we going to have any updates soon?

File: NitroxServer/ConsoleCommands/SaveCommand.cs [hunk 1]
```diff
@@ -1,23 +1,73 @@
...
 
         public override void RunCommand(string[] args, Optional<Player> player)
         {
-            Server.Instance.Save();
+            if((args.Length == 0)||(args[0] == "now")||(args[0] == "-n")){
+                Server.Instance.Save();
+                Log.InfoIG("Server state saved!");
+            }
+            else{
+                if(args[0] == "onlyauto"){
```
Comment (Measurity) [inline NitroxServer/ConsoleCommands/SaveCommand.cs]: Some if branches to similar code. Consider joining the checks together to have less duplicate code:

```
if (args[0] == "test" || args[0] == "other-test") 
{
   // Same code for both cases.
}
```

**Comment by Measurity** [comment]
Why are you merging PRs from other people into your PR? It's not related to your save configuration feature.

**Comment by GuzioMG** [comment]
Well, I merged others PRs, because I thought this repo died. Remember my second comment? Soon after posting it, I noticed, that I can resurrect it. So I changed the name, merged other PRs ...and everything ended there. Now, when someone woke up, I think this is not required anymore. Is there any way to un-merge them?

**Comment by GuzioMG** [comment]
And tomorrow I'll try to do something about all your suggestions! Right now in Poland it's 3:20 AM, so it's not the best time to edit anything, is it?

**Comment by GuzioMG** [comment]
And btw, what is FPSInputModule_ProcessMousePress_Patch.cs? I don't remember that file... I'm not even sure, did I ever opened it. Any ideas?

**Comment by GuzioMG** [comment]
I think I know, why is that. It's caused by my VERY bad idea to put other's suggestions to my branch. So before I screw up anything else, somebody need to explain me how to reject a commit. I need to kill those 7 failures before touching anything.

**Comment by GuzioMG** [comment]
Well... Umm... @Measurity, help. I tried cleaning up that merge mess and I just can't do that. Remember your first review? It was something about the mouse click or similar. Everything else was actually about my changes. But that first one wasn't. It was connected with your comment:

> Why are you merging PRs from other people into your PR? It's not related to your save configuration feature. It becomes less likely that a PR gets merged the bigger it gets.

It came from someone else's repo. And, as you can see on the commits menu, those commits actually disappeared. So my fixing worked, everything is fine, I can now start bringing your suggestions to life and soon it will be merged. Great. ...But not. Changes made by them are still here. And I have no idea, how to fix it. I hope you're a bit more advanced in Git then me and have an idea. Heck, I can't even resolve a simple conflict. Considering, you're the sixth contributor, counting by commits count, you definitely know, how to do that.

**Comment by Measurity** [comment]
Might be better to just copy/paste your work onto a fresh branch from an up-to-date master. If those commits are really gona (or git histroy got messed up another way) then it's probably the best way to go.

**Comment by GuzioMG** [comment]
OK

**Comment by MarijnS95** [comment]
@GuzioMG From the name of the [last commit](https://github.com/SubnauticaNitrox/Nitrox/pull/787/commits/d4f27a837b00437981669c45081eedc68b9df02b) it looks like that's the only change, correct?

If you can validate that all other commits [here](https://github.com/SubnauticaNitrox/Nitrox/pull/787/commits) I'll be able to help you get this updated to `master` without copypasting and _losing commit history info_ which you have neatly done.

**Comment by GuzioMG** [comment]
Just a bit too late...
bitmoji

**Pull request closed without merging**